spec/rdb_backend_spec.rb

```diff
@@ -17,6 +17,9 @@
     it 'supports mysql' do
       expect(Backend::RDBBackend.new(uri, table)).to be_an_instance_of(Backend::RDBBackend)
     end
+    it "accepts ssl_mode as an option" do
+      expect(Backend::RDBBackend.new(uri, table, ssl_mode: :disabled).db.opts[:ssl_mode]).to eq(:disabled)
+    end
```

Comment:
This tests given `ssl_mode` is passed to `Sequel::Mysql2::Database`. But note that it doesn't test `ssl_mode` actually works.
It's nice to confirm passing invalid value to `ssl_mode` raises exception.

Additional to say, how will you confirm PQ's connection is actually using non-SSL connection on testing?

Reply:
I agree trusting `Sequel` is a little risky as it's basically a wrapper like perfectqueue and it could have the same issue, while _in general_ we should not test that layer in unit-testing this library.

Assuming integration test checking what protocol MySQL server received is not easy, for this unit testing purpose, we could at least do something like `expect_any_instance_of(Mysql2::Client).to receive(:ssl_mode=).with(Mysql2::Client::SSL_MODE_DISABLED)`. It'd be worth testing invalid value (as naruse suggested) and missing `ssl_mode` with similar type of assertion.

(FYI: `Mysql2::Client` doesn't seem to have a getter `:ssl_mode`. That's why I suggested to check the argument of the setter)

Reply:
With `show session status like 'ssl_cipher_list%'`, you can confirm whether the connection uses SSL or not.

Also note that by default mysql2 uses UNIX domain socket. To ensure to use TCP, you need to set `host` as `127.0.0.1` instead of `localhost`.
https://github.com/brianmario/mysql2#connecting-to-mysql-on-localhost-and-elsewhere

Reply:
Thank you for the suggestions. I changed the test to confirm session status. 53f3dd8
ssl_mode option is added in mysql 0.4.5, so I also changed gemspec.